SnakeGame/Snake.js

```diff
@@ -0,0 +1,207 @@
+class Snake {
+    constructor(game) {
+        this.game = game;
+        this.direction = "R";
+        this.squeue = new Queue();
+        this.head = [2, 3];
+        // start off as a 3 unit long snake
+        this.squeue.enqueue([2, 1]);
+        this.squeue.enqueue([2, 2]);
+        this.squeue.enqueue([2, 3]);
+        this.appleCount = 0;
+        this.input = "R";
+
+        pageCounter.textContent = "Apples Eaten: " + this.appleCount;
+    }
+
+    move() {
+        let dir = this.direction;
+        if (this.input == "R" && dir == "L"
+            || this.input == "L" && dir == "R"
+            || this.input == "U" && dir == "D"
+            || this.input == "D" && dir == "U") {
+        } else {
+            this.direction = this.input;
+        }
+        dir = this.direction;
+
+        if (dir == "R") {
+            this.squeue.enqueue([this.head[0], this.head[1] + 1]);
+            this.head = [this.head[0], this.head[1] + 1];
+        } else if (dir == "U") {
+            this.squeue.enqueue([this.head[0] - 1, this.head[1]]);
+            this.head = [this.head[0] - 1, this.head[1]];
+        } else if (dir == "L") {
+            this.squeue.enqueue([this.head[0], this.head[1] - 1]);
+            this.head = [this.head[0], this.head[1] - 1];
+        } else if (dir == "D") {
+            this.squeue.enqueue([this.head[0] + 1, this.head[1]]);
+            this.head = [this.head[0] + 1, this.head[1]];
+        }
+        let tail;
+
+        // check if apple is eaten
+        let apple = game.apple;
```

Comment:
Again, making this shorthand (`apple = game.apple`) is not too necessary. I think destructuring assignment is still a better choice here:
```javascript
[hx, hy] = this.head; // update hx and hy since they changed since last time
let [ax, ay] = this.game.apple;
// Now use hx, hy, ax, ay since these are easier to understand than this.head[0], ...etc.
```